src/main/java/core/basesyntax/service/impl/FruitStorageServiceImpl.java

```
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitStorageService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class FruitStorageServiceImpl implements FruitStorageService {
+    private FruitStorageDao fruitStorageDao;
+    private OperationStrategy operationStrategy;
+
+    public FruitStorageServiceImpl(FruitStorageDao fruitStorageDao,
+                                   OperationStrategy operationStrategy) {
+        this.fruitStorageDao = fruitStorageDao;
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void processTransactions(List<FruitTransaction> transactions) {
+        for (FruitTransaction fruitTransaction : transactions) {
+            operationStrategy.get(fruitTransaction.getOperation())
+                    .updateFruitStorage(fruitTransaction, fruitStorageDao);
+        }
+        System.out.println("All transactions processed successfully");
+    }
```

Make sure to handle potential exceptions that could be thrown during the transaction processing. It's important to catch them, wrap them into a custom exception with an informative message, and rethrow, rather than allowing the program to crash or print stack traces.